pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -43,51 +45,72 @@ type ClientCache struct {
 	clientsets    map[unversioned.GroupVersion]*internalclientset.Clientset
 	fedClientSets map[unversioned.GroupVersion]fed_clientset.Interface
 	configs       map[unversioned.GroupVersion]*restclient.Config
-	defaultConfig *restclient.Config
-	// TODO this is only ever read.  It should be moved to initialization at some point.
-	discoveryClient discovery.DiscoveryInterface
-	matchVersion    bool
+
+	defaultConfig     *restclient.Config
+	defaultConfigLock sync.Mutex
```

Comment:
Move this up since it guards defaultConfig and discoveryClient.

matchVersion shouldn't be in this block since it isn't written under the lock.